lib/internal/http2/core.js

```diff
@@ -2377,11 +2385,14 @@ class Http2Server extends NETServer {
   }
 }
 
-function setupCompat(ev) {
+function setupCompat(ServerRequest, ServerResponse, ev) {
   if (ev === 'request') {
     debug('setting up compatibility handler');
     this.removeListener('newListener', setupCompat);
-    this.on('stream', onServerStream);
+    this.on('stream', (stream, headers, flags, rawHeaders) => {
+      onServerStream.call(this, ServerRequest, ServerResponse,
+                          stream, headers, flags, rawHeaders);
```

Comment:
Same as above, you don't need to pass through ServerRequest and ServerResponse. `onServerStream` has access to `this[kOptions]`